src/EditorFeatures/CSharpTest/GenerateFromMembers/GenerateEqualsAndGetHashCodeFromMembers/GenerateEqualsAndGetHashCodeFromMembersTests.cs

```diff
@@ -1608,6 +1721,121 @@ public bool Equals(Program other)
 parameters: CSharp6);
         }
 
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsGenerateEqualsAndGetHashCode)]
+        public async Task TestImplementIEquatableOnClassInNullableContextWithUnannotatedMetadata()
+        {
+            await TestWithPickMembersDialogAsync(
+@"#nullable enable
+
+class Foo
+{
+    public int Bar { get; }
+    [||]
+}",
+@"#nullable enable
+
+using System;
+
+class Foo : IEquatable<Foo?>
+{
+    public int Bar { get; }
+
+    public override bool Equals(object? obj)
+    {
+        return Equals(obj as Foo);
+    }
+
+    public bool Equals(Foo? other)
+    {
+        return other != null &&
```

Comment:
📝  This should have been `other is object` (which is a fairly big problem if `operator ==` was also generated), but if that didn't change in this pull request then it's fine.